helm/platform-health/templates/config_map.yaml

```
@@ -0,0 +1,20 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: platform-health
+  {{- $configMap := .Values.configMap | default dict }}
+  {{- $labels := mergeOverwrite (.Values.common_labels | default dict) ($configMap.labels | default dict) | toYaml | nindent 4 }}
+  labels: {{ $labels }}
+  {{- $annotations := mergeOverwrite (.Values.common_annotations | default dict) ($configMap.annotations | default dict) | toYaml | nindent 4 }}
+  annotations: {{ $annotations }}
+data:
+  {{- $checks := .Values.checks | default dict }}
+  {{- if $checks }}
+  {{ range $provider, $targets := $checks | default dict }}
+  {{- $provider }}: |
+      {{- $targets | toYaml | nindent 6 }}
+  {{ end -}}
+  {{- else }}
+    {}
+  {{- end }}
```

I don't think the data structure is quite right here. Something more like:
```suggestion
data:
  provider-health.yaml: |
  {{- $checks := .Values.checks | default dict }}
  {{- if $checks }}
  {{ range $provider, $targets := $checks | default dict }}
    {{ $provider }}:
      {{ $targets | toYaml | nindent 6 }}
  {{ end -}}
  {{- else }}
    {}
  {{- end }}
```

---

You are absolutely right. :-)